src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/PSUserAgent.cs

```diff
@@ -118,19 +119,36 @@ internal static string App
             }
         }
 
-        internal static string Platform
+        internal static string PlatformName
         {
             get
             {
-                return ("Windows NT");
+                if (Platform.IsWindows)
+                {
+                    return "Windows NT";
```

Comment:
I think we should add tests for these. These could either be a separate test or the existing 'User-Agent` tests could be expanded. 

https://github.com/PowerShell/PowerShell/blob/b07f24e3c9fec8bdeee0f70cb31edbbff0e29b9b/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1#L456

https://github.com/PowerShell/PowerShell/blob/b07f24e3c9fec8bdeee0f70cb31edbbff0e29b9b/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1#L1308

Reply:
The test should be patform sensetive - I'd prefer new separate test.

Reply:
@iSazonov I agree. 
@LDSpits you can use these tests as a base. Please include tests for both `Invoke-WebRequest` and `Invoke-RestMethod`

Reply:
alright, will start work on them later today.

Reply:
@LDSpits Also, it looks like the Windows platform part is supposed to include the `<major>.<minor>` version of windows (6.1, 10.0, etc.). I wouldn't call this a requirement, but if you could add that too it would be great!

Reply:
I think it's better to put it off. If we want more accurate emulation of User-Agent we should discuss in new Issue.

Reply:
It depends on how you look at it. if you view the current issue as "the platform is broken and wrong" then adding the version to the Windows in this PR is correct. After 6.0.0 release I wouldn't recommend changing the platform format until 7.0.0 as changes to it will lead to very undesirable results for users. If we do track that as a separate issue, it should also be 6.0.0-HighPriority.

Reply:
There are a few ways to get the `<major>.<minor>` version of windows:
- Call [`GetVersionEx`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724451(v=vs.85).aspx) via P/Invoke.
- Inspect the `VersionInfo` of a system library. However, the tricky part of this option is to pick the right system binary to inspect because the one you pick might not be available on NanoServer.
```
> $s = Get-Item C:\Windows\System32\kernel32.dll
> $s.VersionInfo.ProductMajorPart
10
> $s.VersionInfo.ProductMinorPart
0
```
- Parse `[RuntimeInformation]::OSDescription`. It should be simple enought too.
```
$osdesp = [System.Runtime.InteropServices.RuntimeInformation]::OSDescription.Trim()
$vString = $osdesp.Substring($osdesp.LastIndexOf(" ")+1)
$version = [version]::new($vString)
$version.Major
$version.Minor
```

Reply:
GetVersionEx looks good. And we could cache the result.

Reply:
@iSazonov I actually personally like parsing `[RuntimeInformation]::OSDescription`. The parsing is a one-time effort and pretty easy. I know you might concern about the consistency of the string value returned from this property ... :)

Reply:
Yes. Ex. - What about Unix? I wouldn't trust `OSDescription`. I'd prefer to discuss all our User-Agent string (properties) before enhance this one.

Reply:
I don't think we need the version info for Linux. This is what I got when using chrome to send request to `http://httpbin.org/user-agent`:
Windows: (Windows NT 10.0; Win64; x64)
Ubuntu: (X11; Linux x86_64)

Reply:
Please see [FireFox](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent/Firefox). Specially MacOS and Android.

Reply:
@daxian-dbw is correct. The OS version numbers in the platform are only sent for Windows platforms. Linux platforms send either `X11` or `Linux` (`Linux` is used more in CLI based HTTP agents). macOS sends `Machintosh`. None of those with versions.